### PR TITLE
My VA - Remove deprecated v0/appointments code

### PR DESCRIPTION
**Pull request by dellerbie**
My VA - Health Care section: Cleanup code post launch of Next Appointment API v2

## Summary

- remove references to the `profile_use_vaos_v2_api` feature toggle
- remove code using the old v0 api

## Related issue(s)
[45480](https://github.com/department-of-veterans-affairs/va.gov-team/issues/45480)
[API PR](https://github.com/department-of-veterans-affairs/vets-api/pull/12750)

## Testing done

- reran unit specs
- tested locally to ensure nothing was broken and only the new v2 api is called

## What areas of the site does it impact?

My VA Dashboard

## Acceptance criteria
- [ ] Toggle is removed
- [ ] Tests are updated

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] Linting warnings have been addressed

### CODE REVIEW

**Comment by adamwhitlock1** [comment]
The appointments v2 cypress tests do seem to be failing within the e2e stress test. I'm going to take a look at this today and see if we can resolve those.

File: src/applications/personalization/dashboard/tests/e2e/appointments-v2-errors.cypress.spec.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+          body: ERROR_400,
+        });
+
+        cy.login(mockUser);
+        cy.visit('my-va/');
+
+        cy.findByText(alertText).should('exist');
+        cy.should(() => {
+          expect(getFacilitiesStub).not.to.be.called;
+        });
```
Comment (adamwhitlock1) [inline src/applications/personalization/dashboard/tests/e2e/appointments-v2-errors.cypress.spec.js:106]: I'm curious about this assertion. I've never used .should directly off `cy` and the docs say that it shouldn't be used this way

https://docs.cypress.io/api/commands/should

I think it's fine for now, but was this something that was copied from the old error e2e test?

**Review by adamwhitlock1 — APPROVED** [review]
I think overall this looks pretty good.

If we are not using the `HealthCare` component anymore and will just use the V2 then should we delete the whole health-care folder?

~~What about the unit tests related to the old component? looks like there are 3 specific unit tests that test the HealthCare component, so maybe those could be refactored to use the v2 version (maybe create a separate ticket for that work)~~ lol just realized your other PR is addressing this, nevermind

File: src/applications/personalization/dashboard/tests/e2e/appointments-v2-errors.cypress.spec.js (hunk 1 above)
Comment (adamwhitlock1) [reply src/applications/personalization/dashboard/tests/e2e/appointments-v2-errors.cypress.spec.js:106]: Yea, go ahead and remove it

**Review by it-harrison — APPROVED** [review]

**Pull request merged**